operator/pkg/clusterconfiguration/expander.go

```diff
@@ -148,7 +150,18 @@ func ExpandForConfiguration(
 			}
 			properties[k] = value
 		case v.ExternalSecretRef != nil:
-			return nil, fmt.Errorf("unimplemented: ExternalSecretRef for configuration entry %q", k)
+			if cloudExpander == nil {
+				return nil, fmt.Errorf("configuration entry %q: external secret references are unsupported", k)
+			}
+			expanded, err := cloudExpander.Expand(ctx, *v.ExternalSecretRef)
+			if err != nil {
+				return nil, fmt.Errorf("configuration entry %q: trouble expanding external secret reference: %w", k, err)
```

Comment:
I haven't found a good linter for this yet so feel free to punt on this comment but upgrading to `errors.Wrapf` from the cockroachdb/errors package will generate much nicer errors here.